### PR TITLE
[primTorch] special: j0, j1, spherical_j0

### PR DESCRIPTION
**Pull request by khushi-411**
Adds prims and refs for special functions (bessel_j0, bessel_j1, spherical_bessel_j0). Thanks!

### CODE REVIEW

**Comment by khushi-411** [comment]
Hi, Could you please help review this PR? Thanks very much!
cc @kshitij12345 @lezcano @mruberry

File: torch/_prims/__init__.py [hunk 1]
```diff
@@ -57,6 +57,8 @@
     "atanh",
     "cos",
     "cosh",
+    "bessel_j0",
```
Comment (mruberry) [inline torch/_prims/__init__.py:60]: Alphabetically these entries go after the `bessel_i` functions

File: torch/_refs/special/__init__.py [hunk 2]
```diff
@@ -17,33 +17,52 @@
...
-def i1(a):
+def i1(a: TensorLikeType) -> TensorLikeType:
     return prims.bessel_i1(a)
 
 
 @_make_elementwise_unary_reference(
     ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT, aten_op=torch.ops.aten.special_i1e
 )
-def i1e(a):
+def i1e(a: TensorLikeType) -> TensorLikeType:
```
Comment (mruberry) [inline torch/_refs/special/__init__.py:65]: Nice typing additions

File: torch/_prims/__init__.py (hunk 1 above)
Comment (khushi-411) [reply torch/_prims/__init__.py:60]: Ooops :p

**Review by mruberry — APPROVED** [review]
LGTM!

**Comment by mruberry** [comment]
It looks like `special.bessel_j0` and `special.bessel_j1` don't render in the current master documentation (`special.spherical_bessel_j0` does). It'd be nice to correct the documentation so they appear. These operations have doc strings:

https://github.com/pytorch/pytorch/blob/e1ed485c65d68787971a2dfb1ace1ed66f5a4d5b/torch/special/__init__.py#L892

but don't appear in `special.rst`: 

https://raw.githubusercontent.com/pytorch/pytorch/master/docs/source/special.rst

**Comment by khushi-411** [comment]
Nice catch! Thanks, @mruberry! Have updated the doc :)

**Comment by khushi-411** [comment]
Nice catch! Thanks, @mruberry! Have updated the doc :)

**Pull request closed without merging**